### ISSUE TITLE
Resource without content files should not be made public

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#2865]
It is possible to create resources without content that are public.  Ways to do this seem to be

As reported by @horsburgh . I used the EnviroDIY website to create this resource:
https://www.hydroshare.org/resource/6e97cc54c6ce448ab32367388a12b793/
Something went wrong in copying the data file over, so there are no content files in this resource. However, the resource is public! 

Also I created a new composite resource, and gave it a title, abstract and keywords, without adding any files.  I get the message that it can be made public.  I can click the public button, and even though I get a warning about it needing file content, it still gets made public.

I was able to do the same with a generic resource.

**Comment by horsburgh** [comment]
According to Craig (@darksinge), he found the error in EnviroDIY that prevented the data file from being copied into the HydroShare resource, but the ability to create a public resource with no content is something that needs to be fixed on the HydroShare side.

**Comment by hyi** [comment]
@dtarb @horsburgh This is a regression since content file was indeed required before a resource could be made public. I have just taken a quick look at the code and this line does not look right to me: https://github.com/hydroshare/hydroshare/blob/e1a11fe20b7a3514e366f66095af9056e158bd10/hs_core/models.py#L2183 Specifically, when a resource does not have supported_upload_file_types defined, like generic resource and possibly composite resource, this has_required_content_files() just returns true which is not right. I believe it should check it at least has files. @aphelionz I can make a fix to this and create a PR against develop which I assume will make its way to release 1.16 if approved and merged. Let me know.

**Comment by hyi** [comment]
@aphelionz I have fixed this issue. I keep that line I indicated above since it is about file types in composite resource and I don't want to remove them without being 100% sure about it. Those lines are not culprit for this issue according to my testing, but the bug was caused by calling a property to check whether there is content file, but that "property" is not defined as property but instead a method. I fixed it by adding property decorator to make it a property. Apparently, this is not captured by tests, so I am going to look into tests to have this test case covered. Since it is a regression, I think it should make into 1.16 release, or even as a hotfix depending on when 1.16 will be released. Let me know which release tag I should create a PR against to have this bug fix along with augmented/added tests included. Looping in @dtarb and @horsburgh to keep them posted as well.

**Comment by hyi** [comment]
@aphelionz I found this is indeed covered by tests. The reason the test did not fail is that the test called this method ```has_required_content_files()``` as a method which is currently being implemented, but ```can_be_public_or_discoverable``` property calls that method ```has_required_content_files``` as a property, hence the issues with UI. I decided to keep this ```has_required_content_files()``` as a method, and fix the bug in ```can_be_public_or_discoverable``` to call it as a method rather than a property. I decided not to add additional unit tests to cover this specific bug, as I believe Selenium UI test will be sure to catch this bug if it props up again in the future. I'll create a PR against 1.16 tag to include this bug fix as part of 1.16 release.
